runtime/core/decoder/onnx_asr_model.h

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 Mobvoi Inc. All Rights Reserved.
+// Author: binbinzhang@mobvoi.com (Binbin Zhang)
+//         di.wu@mobvoi.com (Di Wu)
+// Copyright 2022 Huya Inc. All Rights Reserved.
+// Author: lizexuan@huya.com
+
+#ifndef DECODER_ONNX_ASR_MODEL_H_
+#define DECODER_ONNX_ASR_MODEL_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "./onnxruntime_cxx_api.h"
+#include "decoder/asr_model.h"
+#include "utils/utils.h"
+
+namespace wenet {
+
+class OnnxAsrModel : public AsrModel {
+ public:
+  using TorchModule = torch::jit::script::Module;
+  OnnxAsrModel() = default;
+  OnnxAsrModel(const OnnxAsrModel& other);
+  void Read(const std::string& model_dir, int num_threads = 1);
+  void Reset() override;
+  void AttentionRescoring(const std::vector<std::vector<int>>& hyps,
+                          float reverse_weight,
+                          std::vector<float>* rescoring_score) override;
+  std::shared_ptr<AsrModel> Copy() const override;
+
+ protected:
+  void ForwardEncoderFunc(const std::vector<std::vector<float>>& chunk_feats,
+                          std::vector<std::vector<float>>* ctc_prob) override;
+
+  float ComputeAttentionScore(const float* prob, const std::vector<int>& hyp,
+                              int eos, int decode_out_len);
+
+ private:
+  int encoder_output_size_ = 0;
+  int num_blocks_ = 0;
+  int cnn_module_kernel_ = 0;
+  int head_ = 0;
+
+  Ort::Env env_;
+  std::shared_ptr<Ort::Session> encoder_session_ = nullptr;
+  std::shared_ptr<Ort::Session> rescore_session_ = nullptr;
+  std::shared_ptr<Ort::Session> ctc_session_ = nullptr;
+
+  Ort::MemoryInfo memory_info_ =
+      Ort::MemoryInfo::CreateCpu(OrtDeviceAllocator, OrtMemTypeCPU);
+  std::vector<char*> input_names_ = {
+      "chunk",     "offset",    "required_cache_size",
+      "att_cache", "cnn_cache", "att_mask"};
+  const char* output_names_[3] = {"output", "r_att_cache", "r_cnn_cache"};
+  const char* decode_input_names_[3] = {"hyps", "hyps_lens", "encoder_out"};
+  const char* decode_output_names_[2] = {"score", "r_score"};
+
+  const char* ctc_input_names_[1] = {"hidden"};
+  const char* ctc_output_names_[1] = {"probs"};
+
+  Ort::Value att_cache_ort_{nullptr};
+  Ort::Value cnn_cache_ort_{nullptr};
+  Ort::Value att_mask_ort_{nullptr};
+  std::vector<Ort::Value> encoder_outs_;
+  std::vector<float> att_cache_;
+  std::vector<float> cnn_cache_;
+  std::vector<uint8_t> att_mask_;
```

Comment:
similar to  torch_asr_model, we only need to main `att_cache_ort_` `cnn_cache_ort_` `encoder_outs_` as class members, other variables can be generated on-the-fly.